values.yaml

```diff
@@ -2,31 +2,81 @@
 image:
   name: traefik
   tag: 2.0.5
+
+#
+# Configure the deployment
 #
 deployment:
+  # Number of pods of the deployment
   replicas: 1
+
+rollingUpdate:
+  maxUnavailable: 1
+  maxSurge: 1
+
+# Additional arguments to be passed at the startup of traeffik
+additionalArguments: []
+#  - "--argument1"
+
+#
+# Configure Traefik entrypoints
 #
 ports:
+  # The name of this one can't be changed as it is used for the readiness and
+  # liveness probes, but you can adjust its config to your liking
+  traefik:
+    port: 9000
+    # Defines whether the port is exposed if service.type is LoadBalancer or
+    # NodePort.
+    #
+    # You SHOULD NOT expose the traefik port on production deployments. If you
+    # want to access it from outside of your cluster, use `kubectl proxy` or
+    # create a secure ingress
+    expose: false
+    # The exposed port for this service
+    exposedPort: 9000
   web:
     port: 8000
+    expose: true
+    exposedPort: 80
   websecure:
     port: 8443
-  traefik:
-    port: 9000
-#
-#service:
-#  externalTrafficPolicy: Cluster
+    expose: true
+    exposedPort: 443
+
+# Options for the main traefik service, where the entrypoints traffic comes
+# from.
+service:
+  type: LoadBalancer
+  # Additional annotations (e.g. for cloud provider specific config)
+  annotations: {}
+  # Additional entries here will be added to the service spec. Cannot contains
+  # type, selector or ports entries.
+  spec: {}
```

Comment:
Not entirely sure of that one. We might want to just select the configurable fields ?

Reply:
Good question. WDYT @dtomcej , any advise?

Pro: we let the user able to tune for whatever edge case and don't have to map 1000s of values
Cons: we might have to manage overriding (for instance if end user specifies `type:Cluster IP` here while letting the default `type: Loadbalancer`: key will be duplicated)

Reply:
We have a yaml section in the values.yaml that is applied to the deploment spec.

We may want to move the raw yaml section to each component:

Deployment:
  yaml:

Service:
  yaml:

This would allow users to add custom sections of the spec themselves without having to modify the base chart.

Reply:
although I think that would be best in its own PR

Reply:
I think `yaml` is a bit confusing. It seems to me that it is more standard to see the name of the section where it'll be merged like `spec`